### PR TITLE
feat: support isHit for suggestions

### PR DESCRIPTION
**Pull request by Sherlocksuper**

### CODE REVIEW

File: controllers/message_util.go [hunk 1]
```diff
@@ -224,12 +224,32 @@ func getQuestionWithSuggestions(question string, count int) (string, error) {
...
+		suggestions = append(suggestions, object.Suggestion{Text: formatSuggestion(parts[i]), IsHit: false})
 	}
 
-	return parts[0], parts[1:], nil
+	return parts[0], suggestions, nil
+}
+
+func formatSuggestion(suggestionText string) string {
+	suggestionText = strings.TrimSpace(suggestionText)
+	if strings.HasPrefix(suggestionText, "<") {
```
Comment (hsluoyz) [inline controllers/message_util.go]: TrimPrefix

File: controllers/message_util.go [hunk 2]
```diff
@@ -224,12 +224,32 @@ func getQuestionWithSuggestions(question string, count int) (string, error) {
...
+		suggestionText = suggestionText[1:]
+	}
+	if strings.HasSuffix(suggestionText, ">") {
```
Comment (hsluoyz) [inline controllers/message_util.go]: TrimSuffix

File: web/src/Setting.js [hunk 3]
```diff
@@ -1238,9 +1238,30 @@ export const suggestionsDivider = "|||";
...
 
   return {
     answer: parts[0],
-    suggestions: parts.slice(1),
+    suggestions: suggestions,
   };
 }
+
+export function formatSuggestion(suggestionText) {
+  if (suggestionText.trim().startsWith("<")) {
```
Comment (hsluoyz) [inline web/src/Setting.js]: JS also has trim functions

File: controllers/message_util.go [hunk 4]
```diff
@@ -224,12 +224,32 @@ func getQuestionWithSuggestions(question string, count int) (string, error) {
...
 
+	suggestions := []object.Suggestion{}
+
 	if len(parts) < 2 {
-		return answer, []string{}, nil
+		return answer, suggestions, nil
+	}
+
+	for i := 1; i < len(parts); i++ {
+		suggestions = append(suggestions, object.Suggestion{Text: formatSuggestion(parts[i]), IsHit: false})
```
Comment (hsluoyz) [inline controllers/message_util.go:237]: Why not use prompt to let AI return JSON to you? Avoid do parsing and capsulation by yourself

File: controllers/message_answer.go [hunk 5]
```diff
@@ -228,8 +228,11 @@ func (c *ApiController) GetMessageAnswer() {
...
 	message.Price = modelResult.TotalPrice
 	message.Currency = modelResult.Currency
-
-	textAnswer, textSuggestions, err := parseAnswerAndSuggestions(answer)
+	textAnswer := answer
+	textSuggestions := []object.Suggestion{}
+	if store.SuggestionCount != 0 {
+		textAnswer, textSuggestions, err = parseAnswerAndSuggestions(answer)
+	}
 	if err != nil {
```
Comment (hsluoyz) [inline controllers/message_answer.go]: put into if

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: controllers/message_util.go (hunk 4 above)
Comment (Sherlocksuper) [reply controllers/message_util.go:237]: mean: {suggestion:xxx,isHit:false} ||| {suggestion:xxx,isHit:false} ?
More complex outputs will increase the probability of AI making mistakes
ant "isHit:false"  may waste unnecessary tokens

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**